### DISCUSSION TITLE
new feature: copy to clipboard filtered list

### DISCUSSION THREAD

**Discussion by Ecatonchiro** [ransome1/sleek#290] [Ideas]
More and more I am in a need to filter a list of activities (including completed activities) and send it via email to co-workers. Would it be a good idea to include a "copy to clipboard" shortcut capturing the visibles activities?

**Comment by Ecatonchiro** [comment]
Referring to a similar implementation in orgmode
https://github.com/jkitchin/ox-clip

Referring to the possible API in Electron
https://www.electronjs.org/docs/latest/api/clipboard

**Comment by Ecatonchiro** [comment]
I am new to GitHub and I am taking this project as first attempt to collaborate and understand how a software project works.

**Did you check if a similar feature request has already been reported?**
I found only one request about "copy", however it was included in a long list of other features which took the priority. "Copy" was never implemented or discussed thoroughly. 

**Is your feature request related to a problem? Please describe.**
Currently Sleek cannot communicate with other applications. It is impossible to report project status for instance creating an email with the OPEN/CLOSED activities.

**Describe the solution you'd like**

- [ ] Sleek should allow to copy filtered regions of Todo.txt as **formatted text** on the clipboard that can be pasted into other applications.
- [ ] Sleek should include the OPEN/CLOSED **activities according to the filter** applied on the main page.
- [ ] Sleek should allow the common shortcut Ctrl/CMD-C (currently associated to "create new todo") to copy the activities into the system clipboard.
- [ ] Optionally, Sleek should include in the clipboard a header with the definition of each column (if an optional field is missing, Sleek should include a tab in that field, each field should be separated by a TAB to increase readability)

EXAMPLE - Pasting a filtered activity list - (applied filter: Project=MARKET_1, show completed, sort by due date, priority, project, context)
```
Status	closed_day	priority	started_day	task	Project	Context
			(A)	2021-10-11	Define Legal Plan for	+MARKET_1	@Attorney2 
			(B)	2021-10-22       Polish wording Agreement for       +MARKET_1     @Attorney3 
x	2021-10-29	(A)	2021-10-12	Set meeting with Legal	+MARKET_1	@Attorney1 
```


**Describe alternatives you've considered**
The only possible workaround I can think about is to create a new todo.txt for each new project. Hence I can copy the content of the txt. However this approach will introduce constraints while doing brain dump, and it will make useless the utilization of "+Project" tag.

**Reply by Ecatonchiro** [reply]
Can I ask an opinion to this forum? Do you think this request is valid and can fill a current functionality gap?

**Reply by ransome1** [reply]
> Can I ask an opinion to this forum? Do you think this request is valid and can fill a current functionality gap?

Sorry for not getting back to you. 

I agree with you, that sleek is limited on passing on todo information to other apps and furthermore I think this idea of yours can be very nice addition to sleeks feature list. On that note thanks for your feature requests by the way.

The only issue here is that I can't spend a lot of time on sleek at the moment so I decided to limit development to bugfixing and refactoring mostly. I hope this will change soon. 

As soon as there is work on this I'll let you know. If you know any JavaScript developers or are one yourself, feel free to promote this project, we can really gain a lot from contributions.

**Comment by eterX** [comment]
Hi Everyone,

Would be easier if the output went to a new file? 

I'm thinking `path/to/original/filename-slug-for-the-filter.txt`  or `path/to/original/filename-YYYY-MM-DD-HH-MM-SS.txt` 

Optionally (maybe adding Shift to the shortcut), deleting the original lines.

PD: Sleek rocks, thanks!
